src/Kestrel.Core/Adapter/Internal/AdaptedPipeline.cs

```diff
@@ -79,15 +79,23 @@ public class AdaptedPipeline : IDuplexPipe
                         }
                         else if (buffer.IsSingleSegment)
                         {
+#if NETCOREAPP2_1
+                            await stream.WriteAsync(buffer.First);
+#else
                             var array = buffer.First.GetArray();
                             await stream.WriteAsync(array.Array, array.Offset, array.Count);
```

Comment:
Given that we know `ReadOnlyMemory<byte>.GetArray()` succeeds, what's the benefit of calling `Stream.WriteAsync(ReadOnlyMemory<byte>)`?

Reply:
If the Stream implementation we're calling into doesn't call TryGetArray and wants to pin the memory then it's free here. It also avoids the conversion from Memory\<T\> -> T[] (in theory) in a bunch of places. So for example, when writing through SSL Stream we can avoid the extra pinning required to call into SChannel and OpenSSL.